app/models/host_status/execution_status.rb

```diff
@@ -37,6 +41,47 @@ def to_label(options = {})
     end
   end
 
+  class ExecutionTaskStatusMapper
+    attr_accessor :task
+
+    def self.sql_conditions_for(status)
+      status = STATUS_NAMES.key(status) if status.is_a?(String)
+
+      case status
+        when OK
+          [ "state = 'stopped' AND (result = 'success' OR result = 'warning')" ]
```

Comment:
in rex, warning in dynflow RunHostsJob means mainly that some sub-task has failed - so it's this case (try it). I still thing it would deserve separate state than error (where error is that nothing even started), but it would be still better, than success.